opsdroid/const.py

```diff
@@ -1,15 +1,17 @@
 """Constants used by OpsDroid."""
 import os
+from appdirs import user_log_dir, user_config_dir
 
 __version__ = "0.11.2"
-
+NAME = 'opsdroid'
 DEFAULT_GIT_URL = "https://github.com/opsdroid/"
 MODULES_DIRECTORY = "opsdroid-modules"
 DEFAULT_ROOT_PATH = os.path.expanduser("~/.opsdroid")
-DEFAULT_LOG_FILENAME = os.path.join(DEFAULT_ROOT_PATH, 'output.log')
+DEFAULT_LOG_FILENAME = os.path.join(user_log_dir(NAME, appauthor=False), 'output.log')
 DEFAULT_MODULES_PATH = os.path.join(DEFAULT_ROOT_PATH, "modules")
 DEFAULT_MODULE_DEPS_PATH = os.path.join(DEFAULT_ROOT_PATH, "site-packages")
-DEFAULT_CONFIG_PATH = os.path.join(DEFAULT_ROOT_PATH, "configuration.yaml")
+DEFAULT_CONFIG_PATH = os.path.join(user_config_dir(NAME, appauthor=False), "configuration.yaml")
+OLD_CONFIG_PATH = os.path.join(DEFAULT_ROOT_PATH, "configuration.yaml")
```

Comment:
As time goes by we may forget when this was added and the word `OLD` kinda loses meaning. Perhaps this should be `PRE_0_12_0_CONFIG_PATH` to make it easier to understand in the future.

What do you think?